<launch>

	<node name="dvl_plotter" pkg="sensor_noise_filter" type="plot_dvl.py" output="screen">
		<param name="fs" value="33" />
		<param name="plot_aligned_filtered_signal" value="False" />
	</node>

	<node name="ma_filter_1" pkg="sensor_noise_filter" type="ma_filter.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_1" />
        <param name="window_size" value="5" />
		<param name="label" value="MA, window_size=5" />
	</node>

	<!-- <node name="ma_filter_2" pkg="sensor_noise_filter" type="ma_filter.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_2" />
        <param name="window_size" value="10" />
		<param name="label" value="MA, window_size=10" />
	</node> -->

	<node name="butterworth_1" pkg="sensor_noise_filter" type="butterworth.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_3" />
        <param name="cutoff" value="5" />
		<param name="fs" value="33" />
		<param name="order" value="4" />
		<param name="label" value="butter, cutoff=5,order=4" />
	</node>

	<!-- <node name="butterworth_2" pkg="sensor_noise_filter" type="butterworth.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_4" />
        <param name="cutoff" value="5" />
		<param name="fs" value="33" />
		<param name="order" value="8" />
		<param name="label" value="butter, cutoff=5,order=8" />
	</node> -->

	<node name="ema_filter_1" pkg="sensor_noise_filter" type="ema_filter.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_5" />
        <param name="alpha" value="0.5" />
		<param name="label" value="ema, alpha=.5" />
	</node>

	<!-- <node name="ema_filter_2" pkg="sensor_noise_filter" type="ema_filter.py" output="screen">
		<param name="input_sensor_topic" value="/sensor/dvl" />
        <param name="output_sensor_topic" value="/sensor/dvl_filtered_6" />
        <param name="alpha" value="0.7" />
		<param name="label" value="ema, alpha=.7" />
	</node> -->

</launch>